test/Dialect/Torch/reduce-op-variants.mlir

```diff
@@ -171,3 +184,22 @@ func.func @torch.aten.bernoulli_.float(%t: !torch.tensor) -> !torch.tensor {
   %ret = torch.aten.bernoulli_.float %t, %p, %generator : !torch.tensor, !torch.float, !torch.none -> !torch.tensor
   return %ret : !torch.tensor
 }
+
+// -----
+
```

Comment:
Not sure if this is possible, but can you add a test that shows hows the 4d SDPA op is converted to a 3d linalg_ext with collapse and expands?

Reply:
So the easiest way is to make one of the end-to-end tests do 4D. I had to include the dynamic dimensions cleanup (sorry for stacking).

Reply:
No worries, thank you for accommodating.